src/server.rs

```diff
@@ -411,14 +421,16 @@ async fn detect_context_documents(
 
 async fn detect_generated(
     State(state): State<Arc<ServerState>>,
+    headers: HeaderMap,
     WithRejection(Json(request), _): WithRejection<
         Json<models::DetectionOnGeneratedHttpRequest>,
         Error,
     >,
 ) -> Result<impl IntoResponse, Error> {
     let request_id = Uuid::new_v4();
     request.validate()?;
-    let task = DetectionOnGenerationTask::new(request_id, request);
+    let filtered_headers = filter_header(&state.orchestrator.get_config(), headers).await;
+    let task = DetectionOnGenerationTask::new(request_id, request, filtered_headers);
```

Comment:
Update per other changes.

```suggestion
    let headers = filter_headers(&state.orchestrator.config().passthrough_headers, headers);
    let task = DetectionOnGenerationTask::new(request_id, request, headers);
```